pinot-spi/src/main/java/org/apache/pinot/spi/exception/QueryErrorCode.java

```diff
@@ -57,6 +57,8 @@ public enum QueryErrorCode {
   UNKNOWN_COLUMN(710, "UnknownColumnError"),
   ///  Error while planning the query. For example, trying to run a colocated join on non-colocated tables.
   QUERY_PLANNING(720, "QueryPlanningError"),
+  ///  Query already errored out.
+  ERRORED_OUT(800, "ErroredOutError"),
```

Comment:
I don't think this belongs here. This class represents the error codes sent to the user, but this literal looks like a new value you are adding just to know whether the exception was logged or not. That is not the responsibility of the error code class.

I'm not sure why you need to know if the error was logged or not, but in case you do, it may be a responsibility of the QueryException, but never of the ErrorCode, so maybe you can add a boolean attribute to that class

Reply:
I just added it because felt like where the code was throwing a `RuntimeException`, but I think we can consider the "trying to keep using a mailbox already errored out" situation as an `INTERNAL`.